src/schema/v2/lot.ts

```diff
@@ -0,0 +1,179 @@
+  const { saleArtworksAllLoader, causalityLoader } = context
+  // fetch sale artworks from gravity
+  const { first = 25, ...rest } = args
+
+  const connectionOptions = {
+    include_watched_artworks: true,
+    total_count: true,
+    first,
+    ...rest,
+  }
+  const params = convertConnectionArgsToGravityArgs(connectionOptions)
+  delete params.page
+
+  const { body, headers } = await saleArtworksAllLoader(params)
+  const watchedSaleArtworks: any[] = body
+  const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+  const saleArtworkIds = watchedSaleArtworks.map((sa) => sa._id)
+
+  // Fetch the all fields of the lot object
+  const lotStatesResponse = await causalityLoader({
+    query: gql`
+      query WatchedLotsQuery($ids: [ID!]!) {
+        lots(ids: $ids) {
+          id
+          internalID
+          saleId
+          bidCount
+          reserveStatus
+          sellingPriceCents
+          onlineAskingPriceCents
+          floorSellingPriceCents
+          onlineSellingToBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          floorWinningBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          soldStatus
+        }
+      }
+    `,
+    variables: {
+      ids: saleArtworkIds,
+    },
+  }).then((res: FetchResponse) => res.json())
+
+  const { data: causalityData, errors: causalityErrors } = lotStatesResponse
+
+  // If the causality request failed for some reason, throw its errors.
+  if (causalityErrors) {
+    const errors = causalityErrors.reduce((acc, ce) => {
+      return acc + " " + ce["message"]
+    }, "From causality: ")
+    throw new GraphQLError(errors)
+  } else {
+    // make a map of lot state data
+    const lotDataMap = causalityData.lots.reduce((acc, lot) => {
+      acc[lot.internalID] = lot
+      return acc
+    }, {})
+
+    // place lotData in context for retrieval at stitching stage
+    context.lotDataMap = lotDataMap
+
+    // create nodes from sale artworks
+    const availableLotDataIds = Object.keys(lotDataMap)
+    const nodes = watchedSaleArtworks.reduce((acc, saleArtwork) => {
+      if (!availableLotDataIds.find((id) => id === saleArtwork._id)) {
+        console.warn(`lot state for ${saleArtwork._id} not found - skipping`)
+        return acc
+      } else {
+        return [...acc, { _id: saleArtwork._id, saleArtwork }]
+      }
+    }, [])
+
+    return {
+      totalCount,
+      ...connectionFromArray(nodes, connectionOptions),
+    }
+  }
+}
+
+/**
+ * An extension schema to add the lot type at the causality stitching stage.
+ */
+export const stitchingLotExtensionSchema = gql`
+  # A unified auction lot with data from our auctions bidding engine.
+  extend type Lot {
+    # The current auction state of the lot.
+    lot: AuctionsLotState!
```

Comment:
Here we prepare to extend our Lot type with the `AuctionsLotState` type, which is not yet available to use in the more typical declarative sense at the top of this file, but will be available in causality/stitching.ts.